PIconnect/PI.py

```diff
@@ -79,10 +79,6 @@ def __init__(
         if server and server not in self.servers:
             message = 'Server "{server}" not found, using the default server.'
             warn(message=message.format(server=server), category=UserWarning)
-        if bool(username) != bool(password):
```

Comment:
I think that a username without a password is a valid case, but a password without a user is still invalid, right? In that case we should not remove this check entirely, but make it more similar to the next check for a domain without a username.